src/main/kotlin/org/dashj/platform/dpp/identity/IdentityPublicKey.kt

```diff
@@ -9,27 +9,32 @@ package org.dashj.platform.dpp.identity
 import org.bitcoinj.core.ECKey
 import org.bitcoinj.core.Utils
 import org.dashj.platform.dpp.BaseObject
+import org.dashj.platform.dpp.errors.concensus.signature.InvalidIdentityPublicKeyTypeException
 import org.dashj.platform.dpp.identity.errors.EmptyPublicKeyDataException
 import org.dashj.platform.dpp.toBase64
 import org.dashj.platform.dpp.util.Converters
 
 class IdentityPublicKey(
     val id: Int,
-    val type: TYPES,
+    val type: Type,
     val purpose: Purpose,
     val securityLevel: SecurityLevel,
     val data: ByteArray,
-    val readOnly: Boolean
+    val readOnly: Boolean,
+    var disabledAt: Long? = null,
+    var signature: ByteArray? = null
 ) : BaseObject() {
 
-    enum class TYPES(val value: Int) {
+    enum class Type(val value: Int) {
         ECDSA_SECP256K1(0),
         BLS12_381(1),
-        ECDSA_HASH160(2);
+        ECDSA_HASH160(2),
+        BIP13_SCRIPT_HASH(3),
+        INVALID(30000); // for tests
```

Comment:
`BIP13_SCRIPT_HASH`

There aren't any sign or verify functions that support this type.